src/fastoad/openmdao/problem.py

```diff
@@ -63,11 +64,13 @@ def __init__(self, *args, **kwargs):
     def run_model(self, case_prefix=None, reset_iter_counts=True):
         status = super().run_model(case_prefix, reset_iter_counts)
         ValidityDomainChecker.check_problem_variables(self)
+        BundleLoader().clean_memory()
         return status
 
     def run_driver(self, case_prefix=None, reset_iter_counts=True):
         status = super().run_driver(case_prefix, reset_iter_counts)
         ValidityDomainChecker.check_problem_variables(self)
+        BundleLoader().clean_memory()
```

Comment:
If I comment the two lines (and leave `.clean_memory()` only after the `.setup()`) your test passes. 
If I change your test with a single setup and multiple `run_model`:

```python
def run_problem():
    configurator = oad.FASTOADProblemConfigurator(pth.join(DATA_FOLDER_PATH, "oad_process.yml"))
    configurator.input_file_path = pth.join(RESULTS_FOLDER_PATH, "inputs.xml")
    configurator.output_file_path = pth.join(RESULTS_FOLDER_PATH, "outputs.xml")
    print_memory_state("After reading configuration file")

    ref_inputs = pth.join(DATA_FOLDER_PATH, "CeRAS01.xml")
    configurator.write_needed_inputs(ref_inputs)
    print_memory_state("After writing input file")

    problem = configurator.get_problem(read_inputs=True)
    print_memory_state("After building problem")

    problem.setup()
    print_memory_state("After problem setup")

    count = 100
    for i in range(count):
        print(f"RUN {i+1}/{count}")
        problem.run_model()

    print_memory_state("After problem run")

    problem.write_outputs()
```

The test still passes. Therefore, I wonder if it is necessary to clean the memory after a `run_model`or `run_driver`. Or our tests are not stringent enough? What do you think?